web/src/app/selection/MaterialSelect.js

```diff
@@ -88,12 +98,27 @@ export default class MaterialSelect extends Component {
           }}
           name={name}
           classes={classes}
-          isClearable={!required}
+          isClearable
           isDisabled={disabled}
           isMulti={multiple}
-          value={value}
+          value={isCleared && required ? { label: '', value: '' } : value}
+          clearButtonRef={this.clearButtonRef}
           components={components}
-          onChange={onChange}
+          onBlur={e => {
+            if (
+              isCleared &&
+              e.relatedTarget !== this.clearButtonRef.current &&
+              required
+            ) {
+              this.setState({ isCleared: false })
+              onChange(value)
+            }
+          }}
+          onChange={val => {
+            if (!required) onChange(val)
```

Comment:
It looks like this could fire `onChange` twice, let's invert it so that calling `onChange` is the primary action.

Somethine like:

```js
if (required && val === null) {
    // internally track clear state, without calling onChange if required is true
    this.setState({isCleared: true})
    return
}

onChange(val)
```